src/Host/Broker/Impl/Security/SecurityManager.cs

```diff
@@ -48,6 +49,34 @@ public class SecurityManager {
             return new ClaimsPrincipal(identity);
         }
 
+        private uint CreateProfile(string username, string domain, string password,out StringBuilder profilePath, out bool profileExists) {
+            uint error = 0;
+            profilePath = new StringBuilder();
+            profileExists = false;
+            using (NamedPipeClientStream client = new NamedPipeClientStream("RUserCreatorPipe"))
+            using (BinaryReader reader = new BinaryReader(client))
+            using (BinaryWriter writer = new BinaryWriter(client)) {
+                try {
+                    client.Connect();
+
+                    writer.Write(username);
+                    writer.Write(domain);
+                    writer.Write(password);
+
+                    client.WaitForPipeDrain();
```

Comment:
I'm curious why this is needed. It would seem that the next read should block on receiving the reply anyway?

On the other hand, you might want to explicitly call `Flush` on the writer, since it might have its own buffer.